solidity/contracts/bridge/Bridge.sol

```diff
@@ -778,26 +943,620 @@ contract Bridge is Ownable {
+            uint32 outpointIndex,
             uint256 inputLength
         )
     {
-        inputTxHash = inputVector.extractInputTxIdLeAt(inputStartingIndex);
+        outpointTxHash = inputVector.extractInputTxIdLeAt(inputStartingIndex);
 
-        inputTxIndex = BTCUtils.reverseUint32(
+        outpointIndex = BTCUtils.reverseUint32(
             uint32(inputVector.extractTxIndexLeAt(inputStartingIndex))
         );
 
         inputLength = inputVector.determineInputLengthAt(inputStartingIndex);
 
-        return (inputTxHash, inputTxIndex, inputLength);
+        return (outpointTxHash, outpointIndex, inputLength);
+    }
+
+    /// @notice Requests redemption of the given amount from the specified
+    ///         wallet to the redeemer Bitcoin output script.
+    /// @param walletPubKeyHash The 20-byte wallet public key hash (computed
+    //         using Bitcoin HASH160 over the compressed ECDSA public key)
+    /// @param mainUtxo Data of the wallet's main UTXO, as currently known on
+    ///        the Ethereum chain
+    /// @param redeemerOutputScript The redeemer's length-prefixed output
+    ///        script (P2PKH, P2WPKH, P2SH or P2WSH) that will be used to lock
+    ///        redeemed BTC
+    /// @param amount Requested amount in satoshi. This is also the TBTC amount
+    ///        that is taken from redeemer's balance in the Bank upon request.
+    ///        Once the request is handled, the actual amount of BTC locked
+    ///        on the redeemer output script will be always lower than this value
+    ///        since the treasury and Bitcoin transaction fees must be incurred.
+    ///        The minimal amount satisfying the request can be computed as:
+    ///        `amount - redemptionTreasuryFee - redemptionTxMaxFee`.
+    ///        Fees values are taken at the moment of request creation.
+    /// @dev Requirements:
+    ///      - Wallet behind `walletPubKeyHash` must be active
+    ///      - `mainUtxo` components must point to the recent main UTXO
+    ///        of the given wallet, as currently known on the Ethereum chain.
+    ///      - `redeemerOutputScript` must be a proper Bitcoin script
+    ///      - `redeemerOutputScript` cannot have wallet PKH as payload
+    ///      - `amount` must be above or equal the `redemptionDustThreshold`
+    ///      - Given `walletPubKeyHash` and `redeemerOutputScript` pair can be
+    ///        used for only one pending request at the same time
+    ///      - Wallet must have enough Bitcoin balance to proceed the request
+    ///      - Redeemer must make an allowance in the Bank that the Bridge
+    ///        contract can spend the given `amount`.
+    function requestRedemption(
+        bytes20 walletPubKeyHash,
+        BitcoinTx.UTXO calldata mainUtxo,
+        bytes calldata redeemerOutputScript,
+        uint64 amount
+    ) external {
+        require(
+            wallets[walletPubKeyHash].state == WalletState.Active,
+            "Wallet must be active"
```

Comment:
```suggestion
            wallets[walletPubKeyHash].state == WalletState.Active,
            "Wallet must be in Active state"
```

to align it with the version used in `submitRedemptionProof`: 

https://github.com/keep-network/tbtc-v2/blob/df0fa5378213e180463327bade81073732e6dabd/solidity/contracts/bridge/Bridge.sol#L1172-L1176

Reply:
See https://github.com/keep-network/tbtc-v2/pull/128/commits/b96f1ca4c56abcc4404ff0e78977ed54ff8d30c6